src/UI/App.hs

```diff
@@ -155,7 +155,16 @@ appEvent s (T.AppEvent ev) = case ev of
     then set (asThreadsView . miThreads . listLength) (Just n) s
     else s
   NotifyNewMailArrived n -> M.continue (set (asThreadsView . miNewMail) n s)
-  InputValidated l err -> M.continue (s & set l err . set (asAsync . aValidation) Nothing)
+  InputValidated l err ->
```

Comment:
Wondering if this can be done in a ... well less convoluted way. But currently the best I can come up with.

Reply:
It seems OK to me.  Only suggestions:
- redundant parens in `(Just msg)` pattern
- You could lift the `M.continue` or even `(M.continue . ($ s))` outside the `case` expression, as those are repeated in every branch.
- A bit more commentary about what each case *means* would be nice :)

Reply:
Hm.. interesting that hlint didn't pick up on the redundant parenthesis. Everything else should be done. Cheers!